common/lib/xmodule/xmodule/js/src/sequence/display.coffee

```diff
@@ -40,6 +41,33 @@ class @Sequence
     if position_link and position_link.data('page-title')
         document.title = position_link.data('page-title') + @base_page_title
 
+  hookUpContentStateChangeEvent: ->
+    $('.problems-wrapper').bind(
+      'contentChanged',
+      (event, problem_id, new_content_state) =>
+        @addToUpdatedProblems problem_id, new_content_state
+    )
+
+  addToUpdatedProblems: (problem_id, new_content_state) =>
+    # Used to keep updated problem's state temporarily.
+    # params:
+    #   'problem_id' is problem id.
+    #   'new_content_state' is updated problem's state.
+
+    # initialize for the current sequence if there isn't any updated problem
+    # for this position.
+    if not @anyUpdatedProblems @position
+      @updatedProblems[@position] = {}
+
+    # Now, put problem content against problem id for current active sequence.
+    @updatedProblems[@position][problem_id] = new_content_state
+
+  anyUpdatedProblems:(position) ->
+    # check for the updated problems for given sequence position.
+    # params:
+    #   'position' can be any sequence position.
+    return @updatedProblems[position] != undefined
```

Comment:
@andy-armstrong haha literally the same sequence of thoughts went through my head when I was reviewing this :)